airflow/providers/google/cloud/hooks/gcs.py

```diff
@@ -285,20 +290,43 @@ def download(
         :type chunk_size: int
         :param timeout: Request timeout in seconds.
         :type timeout: int
+        :param num_max_attempts: Number of attempts to download the file.
+        :type num_max_attempts: int
         """
         # TODO: future improvement check file size before downloading,
         #  to check for local space availability
 
-        client = self.get_conn()
-        bucket = client.bucket(bucket_name)
-        blob = bucket.blob(blob_name=object_name, chunk_size=chunk_size)
-
-        if filename:
-            blob.download_to_filename(filename, timeout=timeout)
-            self.log.info('File downloaded to %s', filename)
-            return filename
-        else:
-            return blob.download_as_string()
+        num_file_attempts = 0
+
+        while num_file_attempts < num_max_attempts:
+            try:
+                num_file_attempts += 1
+                client = self.get_conn()
+                bucket = client.bucket(bucket_name)
+                blob = bucket.blob(blob_name=object_name, chunk_size=chunk_size)
+
+                if filename:
+                    blob.download_to_filename(filename, timeout=timeout)
+                    self.log.info('File downloaded to %s', filename)
+                    return filename
+                else:
+                    return blob.download_as_string()
+
+            except GoogleCloudError as e:
```

Comment:
```suggestion
            except GoogleCloudError as err:
```
`e` is not allowed by pylint

Reply:
Thanks - fixed now!